### PR TITLE
[VL] Refactor VeloxParquetDatasource 

### PR DESCRIPTION
**Pull request by gaoyangxiaozhu**
## What changes were proposed in this pull request?

The PR to extend the class to VeloxParquetDatsourceS3, VeloxParquetDatsourceGCS and VeloxParquetDatsourceABFS, insted of puting all of them in the same file as to address this comment https://github.com/apache/incubator-gluten/pull/3978#discussion_r1419928243

and it is also a pre-PR for onboarding **ABFS write part** code change.

(Fixes: \#ISSUE-ID)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by gaoyangxiaozhu** [comment]
@FelixYBW / @tigrux

**Review by tigrux — APPROVED** [review]
Thanks for taking care of this refactor.

File: cpp/velox/operators/writer/VeloxParquetDatasource.h [hunk 1]
```diff
@@ -51,20 +51,31 @@
...
+    }
+  }
+  return false;
+}
+
+inline bool isSupportedGCSPath(const std::string& filePath_) {
+  return strncmp(filePath_.c_str(), "gs:", 3) == 0;
+}
+
+class VeloxParquetDatasource : public Datasource {
```
[the remark this answers is not in the document]
Comment (gaoyangxiaozhu) [reply cpp/velox/operators/writer/VeloxParquetDatasource.h:71]: sure, let me do the change

**Review by JkSelf — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**